processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go

```diff
@@ -56,20 +56,20 @@ func NewDetector(_ component.ProcessorCreateSettings, dcfg internal.DetectorConf
 	return &Detector{metadataProvider: newMetadataClient(sess), tagKeyRegexes: tagKeyRegexes}, nil
 }
 
-func (d *Detector) Detect(ctx context.Context) (pdata.Resource, error) {
+func (d *Detector) Detect(ctx context.Context) (resource pdata.Resource, schemaURL string, err error) {
 	res := pdata.NewResource()
```

Comment:
Wondering if `NewResource` or an overload of it should automatically set schema URL into the resource

Reply:
A sweeping change for all NewResource would be risky, but we can probably introduce an overload.
(It should be actually NewResourceSpans/Metrics/Logs, not NewResource, so something like NewResourceSpansWithSchema).
I don't yet have enough data to see the patterns since this is one a very few components where I added schema support. We can introduce a new method in pdata when we see that there is enough demand for it.